meinberlin/apps/ideas/api.py

```diff
@@ -0,0 +1,300 @@
+from typing import Dict
+from typing import List
+from typing import Tuple
+
+from django.db.models import Model
+from django.db.models import QuerySet
+from django.utils.translation import get_language
+from django.utils.translation import gettext_lazy as _
+from rest_framework import mixins
+from rest_framework import viewsets
+from rest_framework.filters import BaseFilterBackend
+from rest_framework.filters import SearchFilter
+from rest_framework.pagination import BasePagination
+from rest_framework.request import Request
+from rest_framework.response import Response
+from rest_framework.serializers import ModelSerializer
+
+from adhocracy4.api.mixins import ModuleMixin
+from adhocracy4.api.permissions import RulesPermission
+from adhocracy4.api.permissions import ViewSetRulesPermission
+from adhocracy4.categories import get_category_icon_url
+from adhocracy4.categories import has_icons
+from adhocracy4.categories.models import Category
+from adhocracy4.categories.models import CategoryAlias
+from adhocracy4.labels.models import Label
+from adhocracy4.labels.models import LabelAlias
+from adhocracy4.modules.models import Module
+from adhocracy4.modules.predicates import is_allowed_moderate_project
+from meinberlin.apps.contrib.filters import DefaultsRestFilterSet
+from meinberlin.apps.contrib.filters import NoExceptionFilterBackend
+from meinberlin.apps.contrib.filters import OrderingFilterWithDailyRandom
+from meinberlin.apps.moderationtasks.models import ModerationTask
+from meinberlin.apps.moderatorfeedback.models import (
+    DEFAULT_CHOICES as moderator_status_default_choices,
+)
+
+from ..contrib.api import APIPagination
+from .filters import IdeaFilterSet
+from .models import Idea
+from .serializers import IdeaSerializer
+
+
+class LocaleInfoMixin:
+    """Add the current locale of the user to the API response"""
+
+    def list(self, request: Request, *args, **kwargs) -> Response:
+        response = super().list(request, args, kwargs)
+        response.data["locale"] = get_language()
+        return response
```

Comment:
I wonder if this would also go int contrib?

Reply:
@philli-m good point, it definitely could!